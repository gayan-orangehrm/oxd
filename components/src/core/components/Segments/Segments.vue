<script lang="ts">
import {defineComponent, h, PropType} from 'vue';
import {nanoid} from 'nanoid';
import {SEGMENT_DEFAULT} from './types';

export default defineComponent({
  name: 'oxd-segments',
  props: {
    options: {
      type: Array as PropType<any>,
    },
    groupId: {
      type: [String, Number] as PropType<string | number>,
      default: nanoid(8),
    },
    modelValue: {
      type: Object as PropType<any>,
      default: () => SEGMENT_DEFAULT,
    },
  },
  setup() {},
  render() {
    return h(
      'div',
      {
        class: 'oxd-segments-wrapper',
      },
      this.options.map((option: any, i: number) => {
        return h(
          'div',
          {
            class: ['oxd-segment-wrapper', 'd-flex', 'align-center'],
          },
          [
            h(
              'label',
              {
                for: option.id,
                class: [
                  'oxd-segment-label',
                  {
                    active: this.modelValue?.id === option.id,
                  },
                ],
              },
              option.label,
            ),
            h('input', {
              name: this.groupId,
              id: option.id,
              type: 'radio',
              class: 'oxd-segment-input',
              modelValue: this.modelValue,
              onChange: () => this.$emit('update:modelValue', option),
            }),
          ],
        );
      }),
    );
  },
});
</script>

<style src="./segments.scss" lang="scss" scoped></style>
