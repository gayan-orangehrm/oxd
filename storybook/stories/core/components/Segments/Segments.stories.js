import SegmentsStory from './Segments.story.vue';
import {nanoid} from 'nanoid';

export default {
  title: 'Example/Segments',
};

const Template = args => ({
  setup() {
    return {args};
  },
  components: {'oxd-segments-story': SegmentsStory},
  template: '<oxd-segments-story v-bind="args" />',
});

export const Default = Template.bind({});
Default.args = {
  options: [
    {
      id: nanoid(8),
      label: 'Single Assign',
    },
    {
      id: nanoid(8),
      label: 'Bulk Assign',
    },
  ],
};
