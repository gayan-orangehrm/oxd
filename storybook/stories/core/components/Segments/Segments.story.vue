<template>
  <div>
    <p>Selected value: {{ selectedSegment }}</p>
    <oxd-segments
      :options="options"
      v-model="selectedSegment"
    />
  </div>
</template>

<script>
import Segments from '@orangehrm/oxd/core/components/Segments/Segments';
import {nanoid} from 'nanoid';

export default {
  data() {
    return {
      selectedSegment: null,
    };
  },

  props: {
    options: Array,
  },

  components: {
    'oxd-segments': Segments,
  },
};
</script>
