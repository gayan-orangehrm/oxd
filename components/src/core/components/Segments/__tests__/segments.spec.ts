import {mount} from '@vue/test-utils';
import Segments from '@orangehrm/oxd/core/components/Segments/Segments.vue';

describe('Segments > Segments.vue', () => {
  it('renders OXD Segments', () => {
    const wrapper = mount(Segments, {
      props: {
        options: [
          {
            id: 1,
            label: 'Item one',
          },
          {
            id: 2,
            label: 'Item two',
          }
        ]
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
