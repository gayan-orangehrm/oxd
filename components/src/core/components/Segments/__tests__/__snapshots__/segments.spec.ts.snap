// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Segments > Segments.vue renders OXD Segments 1`] = `
<div class="oxd-segments-wrapper">
  <div class="oxd-segment-wrapper d-flex align-center"><label for="1" class="oxd-segment-label">Item one</label><input name="zERZhWrN" id="1" type="radio" class="oxd-segment-input" modelvalue="[object Object]"></div>
  <div class="oxd-segment-wrapper d-flex align-center"><label for="2" class="oxd-segment-label">Item two</label><input name="zERZhWrN" id="2" type="radio" class="oxd-segment-input" modelvalue="[object Object]"></div>
</div>
`;
